sourced/ml/transformers/basic.py

```diff
@@ -300,16 +300,16 @@ def __call__(self, rows: RDD) -> RDD:
         return rows.flatMap(self.deserialize_uast)
 
     def deserialize_uast(self, row: Row):
-        if not row[EngineConstants.Columns.Uast]:
+        if EngineConstants.Columns.Uast not in row:
             return
         row_dict = row.asDict()
         row_dict[EngineConstants.Columns.Uast] = []
         for i, uast in enumerate(row[EngineConstants.Columns.Uast]):
             try:
                 row_dict[EngineConstants.Columns.Uast].append(self.parse_uast(uast))
             except:  # noqa
-                self._log.error("\nBabelfish Error: Failed to parse uast for document %s for uast "
-                                "#%s" % (row[Uast2BagFeatures.Columns.document], i))
+                self._log.error("\nBabelfish Error: Failed to parse uast #%s for repository "
+                                "%s" % (i, row[Uast2BagFeatures.Columns.repository_id]))
```

Comment:
it is always better to have `document` name if it is available. Easier to report a bug. If it is not available we can use `repository_id ` of course. 

Usually, we have  `document` name because we use `UastRow2Document` transformer.